src/entities/user.test.js

```diff
@@ -0,0 +1,28 @@
+import Entity from '../entity'
+import UserEntity from './user'
+
+const data = {
+  id: -1,
+  url: 'https://api.core.ac.uk/internal/users/291',
+  name: 'John Doe',
+  email: 'john.doe@example.com',
+  organisationId: -1,
+  affiliationUrl: 'https://api.example.com/organizations/-1',
+  permissionsUrl: 'https://api.example.com/users/-1/permissions',
+}
+
+describe('affiliation', () => {
+  const user = new UserEntity(data)
+
+  it('exists', () => {
+    expect(user.affiliation).toBeDefined()
+  })
+
+  it('is an Entity', () => {
+    expect(user.affiliation).toBeInstanceOf(Entity)
+  })
+
+  it('has an URL equal to the source', () => {
+    expect(user.affiliation.url).toBe(data.affiliationUrl)
```

Comment:
Which ones and why?

Reply:
I mean for example these two tests. The second one covers use the first test too.
```
  it('exists', () => {
    expect(user.affiliation).toBeDefined()
  })

  it('is an Entity', () => {
    expect(user.affiliation).toBeInstanceOf(Entity)
  })
```

I am not sure whether there can be two expects in one test. If so you could do it like this:

```
  it('Affiliation transformation', () => {
    expect(user.affiliation).toBeInstanceOf(Entity)
    expect(user.affiliation.url).toBe(data.affiliationUrl)
  })
```

Reply:
No, these 2 cover different cases. It's not expressed properly if you do not get it.

The first one check that Entity assigns itself properties (not necessary all) from passed object.

The second one checks transform to be applied.

Third one is probably redundant because it covers some extra logic of Entity which happens only because of configuration.

Thank you for mentioning this. I will create separate PR soon.